mono/sgen/sgen-alloc.c

```diff
@@ -183,6 +183,9 @@ sgen_alloc_obj_nolock (GCVTable vtable, size_t size)
 			g_assert (*p == NULL);
 			mono_atomic_store_seq (p, vtable);
 
+			if (G_UNLIKELY (dump_on_oom_enabled () && !p))
```

Comment:
This condition will always eval to false. In the previous line does `*p = value` meaning p is never NULL